### DISCUSSION TITLE
How long does it take to start contributing?

### DISCUSSION THREAD

**Discussion by woojiq** [servo/servo#30495] [Q&A]
_Asking here, not on zulip, to save it for curious folks._

I'm average developer with no browser engine development experience (only read this [article](https://web.dev/howbrowserswork/)). I want to know the time (week, month, six months, etc.) it takes to start understanding the big picture of a servo, its components, how to write tests... You have worked (probably) with gecko before and it was also something new to you at that time. How long did it take you to start adding features/bug fixes periodically? I found that there are over 3,000 issues, so there's a good chance to find something interesting.
I understand that the correct answer is "it depends on many factors", but a rough average time is enough.

**Comment by mrobinson** [comment]
This is interesting  question, but this is really a better fit for the [Discussions forum](https://github.com/servo/servo/discussions) as the issues here are meant to track features and bugs.

**Reply by sagudev** [reply]
It is possible to convert issues to discussions directly from GH.

**Comment by mrobinson** [comment]
If you have worked on Gecko before, it shouldn't take too long to start working on Servo. The basic design of a web browser doesn't differ very much between browser engines. It is hard to give an exact time period, because this depends entirely on the task you decide to take up. For instance, after poking around a bit you might find a small fix you can make in a few hours. On the other hand, if you are implementing a feature it might take weeks. It really does depend.
